src/main/java/io/confluent/connect/elasticsearch/ElasticsearchClient.java

```diff
@@ -15,102 +15,677 @@
+  private <T> T callWithRetries(String description, Callable<T> function) {
+    try {
+      return RetryUtil.callWithRetries(
+          description,
+          function,
+          config.maxRetries(),
+          config.retryBackoffMs()
+      );
+    } catch (Exception e) {
+      throw new ConnectException("Failed to " + description + ".", e);
+    }
+  }
+
+  /**
+   * Configures HTTP authentication and proxy authentication according to the client configuration.
+   *
+   * @param builder the HttpAsyncClientBuilder
+   */
+  private void configureAuthentication(HttpAsyncClientBuilder builder) {
+
+    CredentialsProvider credentialsProvider = new BasicCredentialsProvider();
+    if (config.isAuthenticatedConnection()) {
+      config.connectionUrls().forEach(
+          url ->
+              credentialsProvider.setCredentials(
+                  new AuthScope(new HttpHost(url)),
+                  new UsernamePasswordCredentials(config.username(), config.password().value())
+              )
+      );
+      builder.setDefaultCredentialsProvider(credentialsProvider);
+    }
+
+    if (config.isBasicProxyConfigured()) {
+      HttpHost proxy = new HttpHost(config.proxyHost(), config.proxyPort());
+      builder.setProxy(proxy);
+
+      if (config.isProxyWithAuthenticationConfigured()) {
+        credentialsProvider.setCredentials(
+            new AuthScope(proxy),
+            new UsernamePasswordCredentials(config.proxyUsername(), config.proxyPassword().value())
+        );
+      }
+
+      builder.setDefaultCredentialsProvider(credentialsProvider);
+    }
   }
 
   /**
-   * Perform a refresh of all indexes, making all indexed data searchable (useful mostly for test)
+   * Configures the client to use SSL if configured.
    *
-   * @throws IOException If the client cannot execute the request
+   * @param builder the HttpAsyncClientBuilder
    */
-  default void refresh() throws IOException {
-    throw new UnsupportedOperationException("refresh is not implemented yet by this client");
+  private void configureSslContext(HttpAsyncClientBuilder builder) {
+    SslFactory sslFactory = new SslFactory(Mode.CLIENT, null, false);
+    sslFactory.configure(config.sslConfigs());
+
+    SSLContext sslContext;
+    try {
+      // try AK <= 2.2 first
+      sslContext = (SSLContext) SslFactory.class.getDeclaredMethod("sslContext").invoke(sslFactory);
+      log.debug("Using AK 2.2 SslFactory methods.");
+    } catch (Exception e) {
+      // must be running AK 2.3+
+      log.debug("Could not find AK 2.2 SslFactory methods. Trying AK 2.3+ methods for SslFactory.");
+
+      Object sslEngine;
+      try {
+        // try AK <= 2.6 second
+        sslEngine = SslFactory.class.getDeclaredMethod("sslEngineBuilder").invoke(sslFactory);
+        log.debug("Using AK 2.2-2.5 SslFactory methods.");
+
+      } catch (Exception ex) {
+        // must be running AK 2.6+
+        log.debug(
+            "Could not find Ak 2.3-2.5 methods for SslFactory."
```

Comment:
"Could not find AK 2.3-2.5 methods for SslFactory."
```suggestion
            "Could not find Ak 2.3-2.5 methods for SslFactory."
```